app/controllers/admin/customers_controller.rb

```diff
@@ -43,6 +43,7 @@ def create
           tag_rule_mapping = TagRule.mapping_for(Enterprise.where(id: @customer.enterprise))
           render_as_json @customer, tag_rule_mapping: tag_rule_mapping
         else
+          set_created_manually_flag if customer_already_exists?
```

Comment:
I think that this is the wrong place. Saving could fail for lots of reasons. I think that we should change line 39:

```
@customer = Customer.new(customer_params)
```

Here we can first check if the customer is already there and re-use that record. The problem then is merging the data. We can ignore the new data and just set the customer to manually created (like you do here) or try to apply the new attributes. I'm not sure what's given in the form already.

If the customer already exists, we could also display a flash message that it was already there and redirect back to the index or edit screen.

Reply:
> If the customer already exists, we could also display a flash message that it was already there and redirect back to the index or edit screen.

I don't see how to redirect to the index.
When I tried to redirect, the API call will be redirected (not the whole page).

Reply:
I don't think we need to redirect here, so I went with displaying the appropriate error. I open a PR here : https://github.com/abdellani/openfoodnetwork/pull/3
The user should be able to close the "add new customer modal" and search for the user they just tried to create if they need to update this user. 

If you actually need to redirect, you would need to handle it here : 
https://github.com/openfoodfoundation/openfoodnetwork/blob/c5ce49e08c4b8377661d7207f0772a5839280f44/app/assets/javascripts/admin/customers/directives/new_customer_dialog.js.coffee#L18-L28

Reply:
> We can ignore the new data and just set the customer to manually created (like you do here) or try to apply the new attributes. I'm not sure what's given in the form already.

We can ignore the new data, as it should be same email that is already in associated with the existing user

Reply:
thank you @rioug :+1: 

I just changes the order of the instructions.

if the customer already exists, we want to set the flag before sending the error.